articles/web-scraping-in-python/index.md

```diff
@@ -0,0 +1,143 @@
+
+#### Getting the data from the web page
+The sample google form that we are going to scrape can be found [**here**](https://docs.google.com/forms/d/e/1FAIpQLSeI8_vYyaJgM7SJM4Y9AWfLq-tglWZh6yt7bEXEOJr_L-hV1A/viewform?formkey=dGx0b1ZrTnoyZDgtYXItMWVBdVlQQWc6MQ)
+
+![Google forms image](/engineering-education/web-scraping-in-python/google-forms.png)
+
+Let us collect all the questions from this form and store them in an excel sheet. For this, we first need to understand the HTML structure of the web page so that we can use beautifulsoup to obtain only the questions. Navigate to the sample google forms website on your browser.
+
+Right-click on any question in the form and select inspect (If you are using Chrome) or inspect element (if you are using firefox).
+
+![Google forms inspect](/engineering-education/web-scraping-in-python/google-forms-inspect.png)
+
+As you can see in the above image, the inspect element allows users to view the HTML contents of the page and make changes to it. It opens a console (right half) where you can view and edit the HTML. We need to obtain the class of the element that contains the questions. Once we get the class, we can use beautifulsoup to get all the elements that have that particular class label and parse them.
+
+![Google forms class](/engineering-education/web-scraping-in-python/google-forms-class.png)
+
+As you can see in the image (highlighted portion), the class we are looking for is called *"freebirdFormviewerComponentsQuestionBaseTitle"*. If we want to obtain the descriptions of the questions (the text below the question), we can use "freebirdFormviewerComponentsQuestionBaseDescription" for the descriptions. However, In this article, we will only scrape the questions.
+
+```python
+def get_data(url):
+
+	# opening the url
+	url_client = urlopen(url)
+	# getting the content of the url.
+	url_content = url_client.read()
+	#close the connection
+	url_client.close()
+	# beautiful soup parser to parse the files in html.
+	parser = soup(url_content, "html.parser")
+	content = parser.findAll('div', {'class': 'freebirdFormviewerComponentsQuestionBaseTitle'})
+	# print(parser.name_of_tag) to print the first element of that particular tag.
+
+	return content
+```
+
+The get_data function takes the URL as a parameter and returns all the elements that belong to a particular class. The urlopen() function establishes a connection to the URL and the read() function reads the contents of that page. We create an object called parser using beautifulsoup. This function takes two parameters: the contents of the page and the type of parser. In this case, we shall use the HTML parser. the findall() method finds all the HTML elements based on a given condition. In this case, we are looking for all the "div" tags that have the class "freebirdFormviewerComponentsQuestionBaseTitle".
```

Comment:
```suggestion
The `get_data` function takes the URL as a parameter and returns all the elements that belong to a particular class. The `urlopen()` function establishes a connection to the URL and the `read()` function reads the contents of that page. We create an object called `parser` using beautifulsoup. This function takes two parameters: the contents of the page and the type of parser. In this case, we shall use the HTML parser. the `findall()` method finds all the HTML elements based on a given condition. In this case, we are looking for all the "div" tags that have the class "freebirdFormviewerComponentsQuestionBaseTitle".
```

Best to wrap any code in code tags so it's clear to the reader that you're referencing functions/variables.